language-server/src/server.js

```diff
@@ -148,43 +179,70 @@ documents.onDidChangeContent(async ({ document }) => {
+  try {
+    const $schema = instance.get("#/$schema");
+   // Determine the context dialect URI from the document
+    contextDialectUri = $schema?.value();
+    dialectUri = contextDialectUri;
+
+    const schemaResources = decomposeSchemaDocument(instance, contextDialectUri);
+    for (const { dialectUri, schemaInstance } of schemaResources) {
+      if (!hasDialect(dialectUri)) {
+        const $schema = schemaInstance.get("#/$schema");
+        if ($schema) {
+          diagnostics.push(buildDiagnostic($schema, "Unknown dialect"));
+        } else {
+          diagnostics.push(buildDiagnostic(schemaInstance, "No dialect"));
+        }
+
+        continue;
       }
 
-      continue;
-    }
-
-    const [output, annotations] = await validate(dialectUri, schemaInstance);
+      const [output, annotations] = await validate(dialectUri, schemaInstance);
 
-    if (!output.valid) {
-      for await (const [instance, message] of invalidNodes(output)) {
-        diagnostics.push(buildDiagnostic(instance, message));
+      if (!output.valid) {
+        for await (const [instance, message] of invalidNodes(output)) {
+          diagnostics.push(buildDiagnostic(instance, message, settings.schemaValidationSeverity === "warning" ? DiagnosticSeverity.Warning : DiagnosticSeverity.Error));
+        }
       }
-    }
 
-    const deprecations = annotations.annotatedWith("deprecated");
-    for (const deprecated of deprecations) {
-      if (deprecated.annotation("deprecated").some((deprecated) => deprecated)) {
-        const message = deprecated.annotation("x-deprecationMessage").join("\n") || "deprecated";
-        diagnostics.push(buildDiagnostic(deprecated.parent(), message, DiagnosticSeverity.Warning, [DiagnosticTag.Deprecated]));
+      const deprecations = annotations.annotatedWith("deprecated");
+      for (const deprecated of deprecations) {
+        if (deprecated.annotation("deprecated").some((deprecated) => deprecated)) {
+          const message = deprecated.annotation("x-deprecationMessage").join("\n") || "deprecated";
+          diagnostics.push(buildDiagnostic(deprecated.parent(), message, DiagnosticSeverity.Warning, [DiagnosticTag.Deprecated]));
+        }
       }
     }
+  } catch (error) {
+    const defaultDialect = settings.defaultDialect;
+    dialectUri = defaultDialect;
   }
 
+
+  connection.console.log(`Schema used: ${dialectUri}`);
   connection.sendDiagnostics({ uri: document.uri, diagnostics });
 };
 
+// Clear cached document settings when configuration changes
+connection.onDidChangeConfiguration((change) => {
+  if (hasConfigurationCapability) {
+    // Reset all cached document settings
+    documentSettings.clear();
+  } else {
+    // Update global settings with new values from configuration change
+    globalSettings = change.settings.jsonSchemaLanguageServer || globalSettings;
+  }
+
+  // Revalidate all open text documents
+  documents.all().forEach((document) => validateSchema(document));
```

Comment:
We're going to need to revalidate the entire workspace, not just the open documents. Look for the `validateWorkspace` function and use that instead.

Reply:
Ahh yes, I overlooked this. makes much more sense to revalidate the entire workspace.